File: pset6/sentimental-readability/readability.py
from cs50 import get_string

texto = get_string("Text: ").strip()
letra, palabras, oraciones = 0,0,0

for i in range(len(texto)):
    if (i == 0 and texto[i] != ' ') or (i != len(texto) - 1 and texto[i] == ' ' and texto[i + 1] != ' '):
        letra += 1
    if texto[i].isalpha():
        palabras += 1
    if texto[i] == '.' or texto[i] == '?' or texto[i] == '!':
        oraciones += 1

L = palabras / letra * 100
S = oraciones / letra * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int grado(int letras, int palabras, int oraciones);

int main(void)
{
    string text = get_string("Text: ");
    int letra = count_letters(text);
    int palabra = count_words(text);
    int oracion = count_sentences(text);

    int Grado = grado(letra, palabra, oracion);
    //dar validacion de grado
    if (Grado >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (Grado < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", Grado);
    }
}

//contador de las letras
int count_letters(string text)
{
    int cuenta = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (!isalpha(text[i]))
        {
            continue;
        }
        else
        {
            cuenta++;
        }
    }
    return cuenta;
}
// contador de espacios, determinas el conteo de palabras
int count_words(string text)
{
    int conteo = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (isspace(text[i]))
        {
            conteo++;
        }
        else
        {
            continue;
        }
    }
    conteo += 1;
    return conteo;
}
//calculando las oraciones determinadas por los signos
int count_sentences(string text)
{
    int contador = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            contador++;
        }
    }
    return contador;
}
// formula del problem set
int grado(int letras, int palabras, int oraciones)
{
    float L = (letras / (float)palabras) * 100;
    float S = (oraciones / (float)palabras) * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    int grado = round(index);
    return grado;
}